src/api/cvc4cpp.cpp

```diff
@@ -2353,5 +2353,234 @@ Sort Solver::declareDatatype(
   return mkDatatypeSort(dtdecl);
 }
 
+/**
+ *  ( declare-fun <symbol> () <sort> )
+ */
+Term Solver::declareFun (const std::string& symbol,
+                         Sort sort) const
```

Comment:
Semi-related to this PR: we could make `Sort` and `Term` const references in these methods (to avoid the overhead of `std::shared_ptr`).

Reply:
Would that apply to more than only the methods in this PR?

Reply:
Yes, this would apply to other methods as well. We have discussed this offline and we came to the conclusion that it is not urgent for now. We can revisit it once we are using the new API.